### ISSUE TITLE
[Defect in 0.4.7] pygit2/pgzip not automatically installed.

### ISSUE THREAD

**Issue by AlexanderVanEck** [facebookresearch/fairscale#1042]
Good morning! Thank you for publishing another hotfix for `fairscale` 🙇‍♂️ I'm sure this will be well received.

# Bug description
Due to the inclusion of entry_points the `pygit2/pgzip` packages are now required by fairscale even though its experimental and optional.

```
entry_points={"console_scripts": ["wgit = fairscale.experimental.wgit.__main__:main"]},
```
https://github.com/facebookresearch/fairscale/blob/main/setup.py#L72

# Steps to reproduce
(in a virtualenv)
```
$ pip install fairscale==0.4.7
$ python -c "import fairscale"
...
Error: please pip install pygit2 module to use wgit
```

# Expected behavior
(in a virtualenv)
```
$ pip install fairscale==0.4.7
$ python -c "import fairscale"
$ # no error
```

# Workaround
(in a virtualenv)
```
$ pip install fairscale!=0.4.7
$ python -c "import fairscale"
$ # no error
```

Would the solution here be to add `pygit2` & `pgzip` to requirements.txt or optionally import them in https://github.com/facebookresearch/fairscale/blob/main/fairscale/experimental/wgit/__init__.py ?

**Comment by anupambhatnagar** [comment]
@min-xu-ai @riohib Can you please have a look at this?

**Comment by min-xu-ai** [comment]
Thanks for the report! I am on it.

**Comment by min-xu-ai** [comment]
Now I see why experimental wgit is imported

```
/home/m1n/git/fairscale/fairscale/experimental/nn/offload.py:435: UserWarning: Failed to initialize NumPy: No module named 'numpy' (Triggered internally at  ../torch/csrc/utils/tensor_numpy.cpp:68.)
  offload_device: torch.device = torch.device("cpu"),
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/m1n/git/fairscale/fairscale/__init__.py", line 12, in <module>
    from . import nn
  File "/home/m1n/git/fairscale/fairscale/nn/__init__.py", line 9, in <module>
    from .data_parallel import FullyShardedDataParallel, ShardedDataParallel
  File "/home/m1n/git/fairscale/fairscale/nn/data_parallel/__init__.py", line 8, in <module>
    from .fully_sharded_data_parallel import FullyShardedDataParallel, OffloadConfig, TrainingState, auto_wrap_bn
  File "/home/m1n/git/fairscale/fairscale/nn/data_parallel/fully_sharded_data_parallel.py", line 54, in <module>
    from fairscale.nn.misc import FlattenParamsWrapper
  File "/home/m1n/git/fairscale/fairscale/nn/misc/__init__.py", line 12, in <module>
    from .flatten_params_wrapper import FlattenParamsWrapper
  File "/home/m1n/git/fairscale/fairscale/nn/misc/flatten_params_wrapper.py", line 35, in <module>
    from fairscale.experimental.nn.ssd_offload import (
  File "/home/m1n/git/fairscale/fairscale/experimental/__init__.py", line 12, in <module>
    from . import nn, optim, wgit
  File "/home/m1n/git/fairscale/fairscale/experimental/wgit/__init__.py", line 9, in <module>
    assert 0
```

**Comment by min-xu-ai** [comment]
fixed in 0.4.8. Will add a test for this going forward as well. Reopen if the issue isn't fixed. Thanks again!
